### PR TITLE
Add SegmentRule models

### LINKED ISSUE #9: Add SegmentRule Models
- [x] SegmentRule model
- [x] SegmentRule model tests
- [x] SegmentRule serlializer
- [x] SegmentRule serlializer tests
- [x] SegmentRuleTask model

As described here:

- https://github.com/onaio/kaznet-web/blob/master/docs/architecture.md
- https://github.com/onaio/kaznet-web/issues/9

### PR DESCRIPTION
**Pull request by moshthepitt**
Fix #9

### CODE REVIEW

File: tasking/models/segment_rules.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    A SegmentRule is basically a way to create a dynamic filter based on
+    the SegmentRule target model.
+
+    For example, if you have this as the SegmentRule:
+        target: Task
+        target_field: id
+        target_field_value: 6
+    Then you would be able to use this SegmentRule to filter:
+        where task_id is 6
```
Comment (urbanslug) [inline tasking/models/segment_rules.py]: should this comment end like so: `SELECT * FROM Task WHERE task_id=6`?

File: tests/serializers/test_segment_rule_serializers.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
```
Comment (urbanslug) [inline tests/serializers/test_segment_rule_serializers.py]: Maybe we should rename this file to "test_segment_rule.py"

File: tests/serializers/test_task_serializers.py [hunk 3]
```diff
@@ -126,6 +126,9 @@ def test_create_task(self):
         now = timezone.now()
         mocked_target_object = mommy.make('tasking.Task')
 
+        rule1 = mommy.make('tasking.SegmentRule')
```
Comment (urbanslug) [inline tests/serializers/test_task_serializers.py]: remove the "_serializers" prefix here too?

**Comment by moshthepitt** [comment]
@urbanslug I addressed your comments

File: tasking/serializers/base.py [hunk 4]
```diff
@@ -43,23 +44,25 @@ class ContentTypeFieldSerializer(serializers.ModelSerializer):
...
+    target_content_type = serializers.PrimaryKeyRelatedField(
+        many=False, queryset=get_allowed_contenttypes())
 
-    target_type = serializers.CharField(
-        source='target_content_type',
-        allow_blank=False)
-    target_app_label = serializers.CharField(
-        write_only=True,
-        allow_blank=False)
+    # target_type = serializers.CharField(
```
Comment (urbanslug) [inline tasking/serializers/base.py]: With version control you shouldn't be afraid to delete code by commenting it out.
You can always retrieve it.

Comment (moshthepitt) [reply tasking/serializers/base.py]: I agree, I think I removed that.

**Pull request merged**